### PR TITLE
feat: temporarily disconnect metadata db during long analytics queries

### PR DESCRIPTION
**Pull request by mistercrunch**
Tried some things ... having issues recreating the scoped session.

Hitting a wall with this message - `Instance <Database at 0xffff7c5305e0> is not bound to a Session; attribute refresh operation cannot proceed (Background on this error at: https://sqlalche.me/e/14/bhk3)`

It's probably just a few lines to restore the database session the right way... Would need to look deeper at the mechanics in https://github.com/pallets-eco/flask-sqlalchemy and reproduce what they do there to re-init the sesh

### CODE REVIEW

File: superset/config.py [hunk 1]
```diff
@@ -204,7 +204,10 @@ def _try_json_readsha(filepath: str, length: int) -> str | None:
 # `SQLALCHEMY_ENGINE_OPTIONS = {"isolation_level": "READ COMMITTED"}`
 # Also note that we recommend READ COMMITTED for regular operation.
 # Find out more here https://flask-sqlalchemy.palletsprojects.com/en/3.1.x/config/
-SQLALCHEMY_ENGINE_OPTIONS = {}
+# for example, if you don't want to use a connection pool, uncomment those next 2 lines
+# from sqlalchemy.pool import NullPool
+# SQLALCHEMY_ENGINE_OPTIONS = {"poolclass": NullPool}
+
```
Comment (eschutho) [inline superset/config.py:210]:
```suggestion
SQLALCHEMY_ENGINE_OPTIONS = {}
```
I'm guessing that you still want the default value.

File: superset/sqllab/sql_json_executer.py [hunk 2]
```diff
@@ -101,6 +102,7 @@ def execute(
             raise
         except Exception as ex:
             logger.exception("Query %i failed unexpectedly", query_id)
+            logger.error(ex)
```
Comment (eschutho) [inline superset/sqllab/sql_json_executer.py:105]: Is this to get some message from the exception logged?

Comment (betodealmeida) [reply superset/sqllab/sql_json_executer.py:105]: `logger.exception` already logs the exception, in addition to the message, it's what differentiates it from `logger.error`. So this line is not needed.

File: superset/models/core.py [hunk 3]
```diff
@@ -95,6 +95,45 @@
...
+    NOTE: only has an effect if feature flag DISABLE_METADATA_DB_DURING_ANALYTICS
+    and using NullPool
+    """
+    pool_type = db.engine.pool.__class__.__name__
+    # Currently only tested/available when used with NullPool
+    do_it = (
+        is_feature_enabled("DISABLE_METADATA_DB_DURING_ANALYTICS")
+        and pool_type == "NullPool"
+    )
+    conn = db.session.connection()
```
Comment (betodealmeida) [inline superset/models/core.py:119]: Do we need this? This is just a new connection, it seems like creating it here, closing it in the `try` and reopening it in the `finally` will do nothing.

**Comment by mistercrunch** [comment]
made the changes, closing in favor of https://github.com/apache/superset/pull/31315

**Pull request closed without merging**